File: packages/api/aqapi/language/domain/value_object/language_code.py
from dataclasses import dataclass
from typing import override

from aqapi.core.domain.value_object.base_value_object import BaseValueObject
from aqapi.core.messages.locale_string import LocaleString

@dataclass
class LanguageCode(BaseValueObject[str]):
    """言語コードの値オブジェクト"""
    
    _value: str

    def __init__(self, value: str):
        super().__init__(value)

    def _validate(self) -> None:
        self._validator.required()

    @property
    @override
    def _value_name(self) -> LocaleString:
        return LocaleString(
            ja="言語コード",
            en="Language Code"
        )

File: packages/flutter/lib/language/domain/value_object/language_code.dart
import 'package:allowance_questboard/core/messages/locale_string.dart';
import 'package:allowance_questboard/core/domain/value_object/base_value_object.dart';

/// 言語コードの値オブジェクト
class LanguageCode extends BaseValueObject<String> {
  LanguageCode(super.value);
  
  @override
  void validate() {
    validator.required();
    validator.integer(); // 言語コードは数値文字列
  }
  
  @override
  LocaleString get valueName => const LocaleString(
    ja: "言語コード",
    en: "Language Code",
  );
}
